test/performance/benchmarks/dataplane-probe/main.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+	"knative.dev/pkg/injection"
+	"knative.dev/serving/test/performance/performance"
+
+	"knative.dev/pkg/signals"
+	"knative.dev/pkg/system"
+)
+
+const (
+	benchmarkName = "Knative Serving dataplane probe"
+)
+
+var (
+	target     = flag.String("target", "", "The target to attack.")
+	duration   = flag.Duration("duration", 5*time.Minute, "The duration of the probe")
+	minDefault = 100 * time.Millisecond
+)
+
+var (
+	// Map the above to our benchmark targets.
+	targets = map[string]struct {
+		target vegeta.Target
+		slaMin time.Duration
+		slaMax time.Duration
+	}{
+		"deployment": {
+			target: vegeta.Target{
+				Method: http.MethodGet,
+				URL:    "http://deployment.default.svc.cluster.local?sleep=100",
+			},
+			// vanilla deployment falls in the +5ms range. This does not have Knative or Istio components
+			// on the dataplane, and so it is intended as a canary to flag environmental
+			// problems that might be causing contemporaneous Knative or Istio runs to fall out of SLA.
+			slaMin: minDefault,
+			slaMax: 105 * time.Millisecond,
+		},
+		"queue": {
+			target: vegeta.Target{
+				Method: http.MethodGet,
+				URL:    "http://queue-proxy.default.svc.cluster.local?sleep=100",
```

Comment:
What happens if we have a long sleep time, does it matter?

Reply:
You mean in that specific test? I haven't tested that.

But in real-traffic-test, if the sleep (=latency of target backend) is too high, and we keep sending some requests per second, the queue gets longer until the client can no longer open new connections.

Reply:
>You mean in that specific test? I haven't tested that.

Yes I was just curious what it means to have more requests around while testing latency variance.
It should not matter in general if it is x+5/10 ms, whatever X is. It should behave like any other golang http server but I was wondering if qp's reverse proxy behaves well in all cases.
For example we set in the transport maxIdleConns=maxIdlePerHost=1000 should be enough but I guess we dont reach that. Btw there were some issues reported on the past https://github.com/golang/go/issues/27816 (probably worth checking with grpc/websockets)? Some thoughts maybe better suited for stress testing.

Reply:
Good ideas for further things to do I think, but maybe a bit far for this PR. Maybe create a new issue for that?

Reply:
Yes this is candidate for future work.